lightning-invoice/src/utils.rs

```diff
@@ -676,6 +720,33 @@ mod test {
 	use crate::utils::create_invoice_from_channelmanager_and_duration_since_epoch;
 	use std::collections::HashSet;
 
+	#[test]
+	fn test_prefer_current_channel() {
+		let test_cases = vec![
+			// No minimum, prefer larger candidate channel.
+			(None, 100, 200, false),
+			// No minimum, prefer larger current channel.
+			(None, 200, 100, true),
+			// Minimum set, prefer current channel over minimum + buffer.
+			(Some(100), 115, 100, true),
+			// Minimum set, prefer candidate channel over minimum + buffer.
+			(Some(100), 105, 125, false),
+			// Minimum set, both channels sufficient, prefer smaller current channel.
+			(Some(100), 115, 125, true),
+			// Minimum set, both channels sufficient, prefer smaller candidate channel.
+			(Some(100), 200, 160, false),
+			// Minimum set, neither sufficient, prefer larger current channel.
+			(Some(200), 100, 50, true),
+			// Minimum set, neither sufficient, prefer larger candidate channel.
+			(Some(200), 100, 150, false),
+		];
+
+		for test_case in test_cases {
+			assert_eq!(crate::utils::prefer_current_channel(test_case.0, test_case.1, test_case.2), test_case.3);
+		}
```

Comment:
I'd prefer that we don't use a loop here. Otherwise, it isn't clear which case has failed.